src/test/fuzz/util.h

```diff
@@ -35,6 +35,17 @@
 #include <string>
 #include <vector>
 
+template <typename... Callables>
+void CallOneOf(FuzzedDataProvider& fuzzed_data_provider, Callables... callables)
+{
+    constexpr size_t call_size{sizeof...(callables)};
+    static_assert(call_size >= 1);
+    const size_t call_index{fuzzed_data_provider.ConsumeIntegralInRange<size_t>(0, call_size - 1)};
+
+    size_t i{0};
```

Comment:
Note: I couldn't figure out how to make `i` evaluate at compile time, but I presume that `-O1` already flattens this out.

Reply:
turns out `-O1` isn't enough for clang, but `-O2` seems to do it: https://godbolt.org/z/Pz1YGv

Reply:
You can evaluate `i` at compile time with more recursive templates:

```c++
template <size_t i>
inline void CallNthOf(size_t n) { }

template <size_t i, typename Callable, typename... Callables>
inline void CallNthOf(size_t n, Callable callable, Callables... callables)
{
    if (i == n) callable();
    CallNthOf<i+1>(n, callables...);
}
```

and then do `return CallNthOf<0>(call_index, callables...)`. But it looks to me like `clang -O1` doesn't even do inlining, so doesn't seem like there's any reason to poke at this further?

Very impressed that you got it equivalent to a switch though!!

Reply:
Closing discussion for now because there is nothing left to do right now